### ISSUE TITLE
discord bot integration

### ISSUE THREAD

**Issue by twk-b** [a-tal/irace#11]
# discord bot integration

messages to CHANNELX based on new member join

* iracing data lookup, pull bios for new members, 
* verify name and other details
* verify they are a member of Club Canada
* enforce other policies??

message to CHANNELX based on race schedule

* alerts you for any upcoming league races race
    * 9AM on day of, 
    * 1hr before
    * 5mins to reg ending

message to CHANNELX on based Racer completing:

* reports qually lap time and rank as race starts
* reports finish position, race time, fastest lap, incidents

message to CHANNELX on command

* /driver ##### - pull driver info
* /club #### pull club info
* /series ### pull series info
* /calendar  show upcoming events

## how

* itwkracing pulls data from iracingSDK while race is running
* itwkracing reports events to itwkracing.dcbl.ca
* itwkracing.dcbl.ca buffers events, dedupping events from same event.... 
* itwkracing.dcbl.ca uses a-tal/irace to pull stats for message
* itwkracing.dcbl.ca messages discord with message

## additional info

* bot already created has some basic stuff working
    * can do a vote in channel
    * cant remember what else, will update later

**Comment by a-tal** [comment]
I don't think there's a way to pull qualify times while the race is still ongoing. if you know of a way let me know though

**Comment by twk-b** [comment]
i have them probably as soon as the Q session closes, maybe when the race session starts... its included in the SI.json data (deoending when that snapshot was taken)....  not from website, pulling from iracingSDK from telemetry.... SI gets updated approximately every time somene finishs a lap...

**Comment by a-tal** [comment]
fair enough. the sdk is out of scope for this repo though, we only deal with web stuff here
